TeachingRecordSystem/tests/TeachingRecordSystem.TestCommon/CrmTestData.UpdatePerson.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.Xrm.Sdk.Messages;
+using TeachingRecordSystem.Core.Dqt.Models;
+
+namespace TeachingRecordSystem.TestCommon;
+
+public partial class CrmTestData
+{
+    public Task UpdatePerson(Action<UpdatePersonBuilder>? configure)
+    {
+        var builder = new UpdatePersonBuilder();
+        configure?.Invoke(builder);
+        return builder.Execute(this);
+    }
+
+    public class UpdatePersonBuilder
+    {
+        Guid? _personId = null;
+        (string FirstName, string MiddleName, string LastName)? _updatedName = null;
+
+        public UpdatePersonBuilder WithPersonId(Guid personId)
+        {
+            if (_personId is not null)
+            {
+                throw new InvalidOperationException("WithPersonId has already been set");
+            }
+
+            _personId = personId;
+            return this;
+        }
+
+        public UpdatePersonBuilder WithUpdatedName(string firstName, string middleName, string lastName)
+        {
+            if (_updatedName is not null)
+            {
+                throw new InvalidOperationException("WithUpdatedName has already been set");
+            }
+
+            _updatedName = (firstName, middleName, lastName);
+            return this;
+        }
+
+        public async Task Execute(CrmTestData testData)
+        {
+            if (_personId is not null && _updatedName is not null)
```

Comment:
I think we should throw here if `_personId is null`

Reply:
Done